core/node/dns.go

```diff
@@ -0,0 +1,18 @@
+package node
+
+import (
+	config "github.com/ipfs/go-ipfs-config"
+	doh "github.com/libp2p/go-doh-resolver"
+	madns "github.com/multiformats/go-multiaddr-dns"
+)
+
+func DNSResolver(cfg *config.Config) (*madns.Resolver, error) {
+	var opts []madns.Option
+	if cfg.DNS.DefaultResolver != "" {
+		opts = append(opts, madns.WithDefaultResolver(doh.NewResolver(cfg.DNS.DefaultResolver)))
+	}
+	for domain, url := range cfg.DNS.CustomResolvers {
+		opts = append(opts, madns.WithDomainResolver(domain, doh.NewResolver(url)))
```

Comment:
the `net.Resolver` type doesn't offer a public constructor unfortunately.

Reply:
Looks like you can create a resolver struct with a custom Dialer https://stackoverflow.com/questions/59889882/specifying-dns-server-for-lookup-in-go.

Either way though we need some syntax for how we describe the addresses as standard vs DoH.

cc @lidel for thoughts. We could just use plain IP address vs HTTPS url. It could be nice to use multiaddrs since they're more uniform, but I seem to recall we ran into problems there last time (perhaps they aren't blockers though?).

Reply:
Let's go with https urls -- and for now we can just support DoH and error if the url is not an https url.

Reply:
Added check for urls starting with `https://` before creating a DoH resolver.
When we have more resolvers or decide we want to use something else, we can easily add support by checking the url.